nlp/2021_02_26_compact_transformers/compact_transformers.ipynb

```diff
@@ -0,0 +1,3861 @@
+            "top": null,
+            "grid_column": null,
+            "overflow_y": null,
+            "overflow_x": null,
+            "grid_auto_flow": null,
+            "grid_area": null,
+            "grid_template_columns": null,
+            "flex": null,
+            "_model_name": "LayoutModel",
+            "justify_items": null,
+            "grid_row": null,
+            "max_height": null,
+            "align_content": null,
+            "visibility": null,
+            "align_self": null,
+            "height": null,
+            "min_height": null,
+            "padding": null,
+            "grid_auto_rows": null,
+            "grid_gap": null,
+            "max_width": null,
+            "order": null,
+            "_view_module_version": "1.2.0",
+            "grid_template_areas": null,
+            "object_position": null,
+            "object_fit": null,
+            "grid_auto_columns": null,
+            "margin": null,
+            "display": null,
+            "left": null
+          }
+        }
+      }
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4Hp3Msndwc3y"
+      },
+      "source": [
+        "## Compact Transformers  🤏\n",
+        "\n",
+        "We all know bigger is often better! Nothing cooler than rolling up in your larger-than-life Transformer model. Who doesn't want to drive a monster truck once in their life...\n",
+        "\n",
+        "However, monster truck aren't always practical in real life. Models need to run smoothly in production. So the best model size is often the one most suited for your practical situation and requirements.\n",
+        "\n",
+        "In this tip, we'll have a look at some sub-25million parameter transformer models available in Huggingface. So you can have both a smoothly running application and brag to your manager that you implemented transformers!\n",
+        "\n",
+        "### The test 🔨\n",
+        "We will be evaluating all of our candidates on a few fronts:\n",
+        "\n",
+        "\n",
+        "*   GPU memory during finetune: relevant in case of hardware or cost limits\n",
+        "*   GPU memory during inference: relevant when running on edge device\n",
+        "*   Model artefact size: relevant when facing hardware restrictions\n",
+        "*   Model CPU & GPU inference time: relevant when dealing with latency restrictions\n",
+        "*   Time taken to finetune: relevant when dealing with frequent retraining loops\n",
+        "\n",
+        "All of these benchmarked against the final performance after training for one epoch!\n",
+        "\n",
+        "*Note: tests were done in Google Colab, with a V100 GPU.*\n",
+        "\n",
+        "... let the games begin 🏋️‍♀️\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-rcrjcbtxqgB"
+      },
+      "source": [
+        "### Setup"
```

Comment:
Could maybe be nice to put a `nvidia-smi` here, to check whether there's a GPU.

Reply:
That's a good point, will add